src/lib/krb5/os/dnssrv.c

```diff
@@ -45,6 +45,122 @@ krb5int_free_srv_dns_data (struct srv_dns_entry *p)
     }
 }
 
+static krb5_error_code
+prepare_lookup_buf(const krb5_data *realm, const char *service,
+                   const char *protocol, char *fixed_buf, size_t bufsize)
+{
+    struct k5buf buf;
+    if (memchr(realm->data, 0, realm->length))
+        return EINVAL;
+
+    k5_buf_init_fixed(&buf, fixed_buf, bufsize);
+    k5_buf_add_fmt(&buf, "%s.", service);
+    if (protocol != NULL)
+        k5_buf_add_fmt(&buf, "%s.", protocol);
+    k5_buf_add_len(&buf, realm->data, realm->length);
+
+    /* Realm names don't (normally) end with ".", but if the query
```

Comment:
This is of course moved code, but please reformat the multi-line comment according to current practice.  Also, please drop the second paragraph of the comment.  Nothing in the code checks for "only if a name is specified," and we never make SRV or URI queries on a prefix alone.  I tracked down the history of this comment and we didn't check for "only if a name is specified" back then either, so I don't know why that paragraph was added.